server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -0,0 +1,468 @@
+        return false;
+    }
+
+    @Override
+    public void writeIndexingBuffer() throws EngineException {}
+
+    @Override
+    public boolean shouldPeriodicallyFlush() {
+        return false;
+    }
+
+    @Override
+    public void flush(boolean force, boolean waitIfOngoing) throws EngineException {}
+
+    @Override
+    public void trimUnreferencedTranslogFiles() throws EngineException {
+        ensureOpen();
+        try {
+            translog.trimUnreferencedReaders();
+        } catch (IOException e) {
+            throw new EngineException(shardId, "failed to trim translog", e);
+        }
+    }
+
+    @Override
+    public boolean shouldRollTranslogGeneration() {
+        return translog.shouldRollGeneration();
+    }
+
+    @Override
+    public void rollTranslogGeneration() throws EngineException {
+        ensureOpen();
+        try {
+            translog.rollGeneration();
+            translog.trimUnreferencedReaders();
+        } catch (IOException e) {
+            throw new EngineException(shardId, "failed to roll translog", e);
+        }
+    }
+
+    @Override
+    public void forceMerge(
+        boolean flush,
+        int maxNumSegments,
+        boolean onlyExpungeDeletes,
+        boolean upgrade,
+        boolean upgradeOnlyAncientSegments,
+        String forceMergeUUID
+    ) throws EngineException, IOException {}
+
+    @Override
+    public GatedCloseable<IndexCommit> acquireLastIndexCommit(boolean flushFirst) throws EngineException {
+        try {
+            final IndexCommit indexCommit = Lucene.getIndexCommit(lastCommittedSegmentInfos, store.directory());
+            return new GatedCloseable<>(indexCommit, () -> {});
+        } catch (IOException e) {
+            throw new EngineException(shardId, "Unable to build latest IndexCommit", e);
+        }
+    }
+
+    @Override
+    public GatedCloseable<IndexCommit> acquireSafeIndexCommit() throws EngineException {
+        return acquireLastIndexCommit(false);
+    }
+
+    @Override
+    public SafeCommitInfo getSafeCommitInfo() {
+        return new SafeCommitInfo(localCheckpointTracker.getProcessedCheckpoint(), lastCommittedSegmentInfos.totalMaxDoc());
+    }
+
+    @Override
+    protected final void closeNoLock(String reason, CountDownLatch closedLatch) {
+        if (isClosed.compareAndSet(false, true)) {
+            assert rwl.isWriteLockedByCurrentThread() || failEngineLock.isHeldByCurrentThread()
+                : "Either the write lock must be held or the engine must be currently be failing itself";
+            try {
+                IOUtils.close(readerManager, translog, store::decRef);
+            } catch (Exception e) {
+                logger.warn("failed to close engine", e);
+            } finally {
+                try {
+                    logger.debug("engine closed [{}]", reason);
+                } finally {
+                    closedLatch.countDown();
+                }
+            }
+        }
+    }
+
```

Comment:
Do we need the try block?

Reply:
Nope, thanks.  I originally had the store.decref() in that block but moved it up.